VideoLocker/src/test/java/org/edx/mobile/test/http/ExtensionRegistryTest.java

```diff
@@ -0,0 +1,35 @@
+package org.edx.mobile.test.http;
+
+import org.edx.mobile.view.ExtensionRegistry;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.mock;
+
+public class ExtensionRegistryTest {
+
+    ExtensionRegistry extensionRegistry;
+
+    @Before
+    public void setup() {
+        extensionRegistry = new ExtensionRegistry();
+    }
+
+    @Test
+    public void forType_withNoExtensions_returnsEmptyRegistry() {
+        assertThat(extensionRegistry.forType(TestExtension.class).iterator().hasNext(), is(false));
+    }
+
+    @Test
+    public void forType_withAddedExtension_returnsRegistryWithAddedExtension() {
+        final TestExtension extension = mock(TestExtension.class);
+        extensionRegistry.forType(TestExtension.class).add(extension);
+        assertThat(extensionRegistry.forType(TestExtension.class).iterator().hasNext(), is(true));
+        assertThat(extensionRegistry.forType(TestExtension.class).iterator().next(), is(extension));
+    }
+
+    public interface TestExtension extends ExtensionRegistry.Extension {
```

Comment:
Maybe a test where if you add two registries for different types they go to separate places?